backend/background/util/elastic.py

```
@@ -51,4 +78,53 @@ def verify_elasticsearch():
+
+
+def ensure_elasticsearch_index() -> bool:
+    """
+    Ensure the Elasticsearch index exists with the expected vector store mappings.
+    """
+    try:
+        exists = es_client.indices.exists(index=ELASTIC_INDEX)
+        if exists.meta.status == 200:
+            logger.info(f"Elasticsearch index {ELASTIC_INDEX} already exists")
+            return True
+    except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)